File: Q4.py
def anagrams(word, word_list):
    word_sorted = sorted(word.lower().replace(" ", ""))
    anagrams_list = []
    for w in word_list:
        if sorted(w.lower().replace(" ", "")) == word_sorted:
            anagrams_list.append(w)
    return anagrams_list
#Example
#word = "listen"
#word_list = ["enlists", "google", "inlets", "banana"]
#anagrams_list = anagrams(word, word_list)
#print(anagrams_list) # Output: ["enlists", "inlets"]

File: Q4.cpp
#include <iostream>
#include <algorithm>
using namespace std;

bool isAnagram(string word1, string word2) {
    sort(word1.begin(), word1.end());
    sort(word2.begin(), word2.end());
    return word1 == word2;
}
int main() {
    string word = "listen";
    string
word_list[] = {"enlists", "google", "inlets", "banana"};
int n = sizeof(word_list) / sizeof(word_list[0]);
for (int i = 0; i < n; i++) {
if (isAnagram(word, word_list[i])) {
cout << word_list[i] << " ";
}
}
return 0;
}
